src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -1054,6 +1189,62 @@ bool QgsMapToolLabel::labelMoveable( QgsVectorLayer *vlayer, const QgsPalLayerSe
   return false;
 }
 
+bool QgsMapToolLabel::labelAnchorPercentMovable( QgsVectorLayer *vlayer, const QgsPalLayerSettings &settings, int &lineAnchorPercentCol, int &lineAnchorClippingCol,  int &lineAnchorTypeCol,  int &lineAnchorTextPointCol ) const
+{
+  lineAnchorPercentCol = -1;
+  lineAnchorClippingCol = -1;
+  lineAnchorTypeCol = -1;
+  lineAnchorTextPointCol = -1;
+
+  bool ret { true };
+
+  if ( settings.dataDefinedProperties().isActive( QgsPalLayerSettings::LineAnchorPercent ) )
```

Comment:
suggestion: a lambda function could be used to factor a bit that instance and the 3 following similar ones.